medschool/src/main.rs

```diff
@@ -0,0 +1,550 @@
+        // belong to other types
+        .flat_map(|syntaxed_file| {
+            syntaxed_file
+                .items
+                .into_iter()
+                // convert an `Item` into a `PartialType`
+                .filter_map(|item| match item {
+                    syn::Item::Mod(item_mod) => {
+                        docs_from_attributes(item_mod.attrs);
+                        None
+                    }
+                    syn::Item::Enum(item) => {
+                        let thing_docs_untreated = docs_from_attributes(item.attrs);
+                        let is_internal = thing_docs_untreated
+                            .iter()
+                            .any(|doc| doc.contains(r"<!--${internal}-->"));
+
+                        // We only care about types that have docs.
+                        if !thing_docs_untreated.is_empty() && !is_internal {
+                            Some(PartialType {
+                                ident: item.ident.to_string(),
+                                docs: thing_docs_untreated,
+                                fields: Default::default(),
+                            })
+                        } else {
+                            None
+                        }
+                    }
+                    syn::Item::Struct(item) => {
+                        let mut fields = item
+                            .fields
+                            .into_iter()
+                            .filter_map(PartialField::new)
+                            .collect::<HashSet<_>>()
+                            .into_iter()
+                            .collect::<Vec<_>>();
+                        fields.sort();
+
+                        let thing_docs_untreated = docs_from_attributes(item.attrs);
+                        let is_internal = thing_docs_untreated
+                            .iter()
+                            .any(|doc| doc.contains(r"<!--${internal}-->"));
+
+                        let public_fields = fields
+                            .into_iter()
+                            .filter(|field| !field.docs.contains(&r"<!--${internal}-->".into()))
+                            .collect::<Vec<_>>();
+
+                        // We only care about types that have docs.
+                        if !thing_docs_untreated.is_empty()
+                            && !public_fields.is_empty()
+                            && !is_internal
+                        {
+                            Some(PartialType {
+                                ident: item.ident.to_string(),
+                                docs: thing_docs_untreated,
+                                fields: public_fields,
+                            })
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                })
+        })
+        .collect::<BTreeSet<_>>();
+
```

Comment:
can you please document the reason why you use btreeset/btreemap over hashmap

Reply:
Yup, can you spot more places where docs are lacking?

Will open a PR with whatever needs tidying up.

Reply:
sure will let you know

Reply:
Added a brief doc for this in the newest PR.